filters/special_casing/filter.py

```diff
@@ -0,0 +1,20 @@
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class TextSpecialCasingFilter(SentenceOperation):
+    """Filter if any of the special casings appear in the sentence:
+    All text is capitalized or All text is lowercased or there is Title Casing in the Sentence.
+    """
+
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+
+    def filter(self, sentence: str = None) -> bool:
+
+        return any(
+            [sentence.isupper(), sentence.islower(), sentence.istitle()]
```

Comment:
Why not also check for Title Case and CamelCase as well? Also, how about adding a parameter to select which casings to test for?

Reply:
Thanks, the filter checks for title casing already. Also, camel casing is for words, right? Is there a sequence analogue to camel casing?

Reply:
I think it makes sense from a modularity perspective to have one configurable filter than three separate static filters, but I'm open to arguments. 

Are the types of casing actually mutually exclusive within a sequence though? It seems to me that you could have an input that includes multiple types, e.g. (the following is a STRANGE WAY to Type Words Doesn'tItSeem?)

Oh, and on camel case, it's just any sequence without spaces where the word boundary is instead indicated by capitalisation, so can be arbitrary length.

Reply:
I second on the issue of the hybrid "weird" cases. Following the logic of combining several types of casing together such hybrid cases should be filtered as well. 
Also, ideally one would like to take into account the fact that some words such as proper nouns might used capital letter in the middle of the sentence but they should not be considered as "weird" hybrid cases.

Reply:
Thanks @richplant , @denkle for the comments. Regarding weird "hybrid" cases, e.g. "STRANGE WAY to Type Words Doesn'tItSeem?" and "MICHAEL JACKSON charged in Nevada, U.S.A" or "Lion King a massive success", there is no generic logic that will be able to tell apart the "correct" from the "hybrid" weird, right? That's why the proposed filter works only at the sequence level currently. That doesn't cover all "weird" cases, but whatever it covers (all lowercase, all uppercase, title case), it does so with perfect precision.

Reply:
> Are the types of casing actually mutually exclusive within a sequence though? It seems to me that you could have an input that includes multiple types, e.g. (the following is a STRANGE WAY to Type Words Doesn'tItSeem?)

Yes, the cases in the filter are mutually exclusive ( a sentence can be either all lowercased, all uppercased or title-cased). The reason for not getting into the hybrid "weird" cases is that they don't have a consistent logic to exploit. I have an example of why that is in the above comment.

> Oh, and on camel case, it's just any sequence without spaces where the word boundary is instead indicated by capitalisation, so can be arbitrary length.

Yes, but that is for words (IsThisAGoodExample), right not sentences? In languages, barring a few, we typically expect white-space tokenized sentences. The sequence (sentence) analogue of camel case is Title Casing (e.g. "Germany Wins World Cup"), which is already covered.